EFCore.CheckConstraints/CheckConstraints/Internal/DiscriminatorCheckConstraintConvention.cs

```diff
@@ -0,0 +1,52 @@
+using System.Linq;
+using System.Text;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Conventions;
+using Microsoft.EntityFrameworkCore.Storage;
+
+namespace EFCore.CheckConstraints.Internal
+{
+    /// <summary>
+    ///     A convention that creates check constraints ensuring that (complete) discriminator columns only have
+    ///     expected values.
+    /// </summary>
+    public class DiscriminatorCheckConstraintConvention : IModelFinalizingConvention
+    {
+        readonly ISqlGenerationHelper _sqlGenerationHelper;
+
+        public DiscriminatorCheckConstraintConvention(ISqlGenerationHelper sqlGenerationHelper)
+            => _sqlGenerationHelper = sqlGenerationHelper;
+
+        /// <inheritdoc />
+        public virtual void ProcessModelFinalizing(IConventionModelBuilder modelBuilder, IConventionContext<IConventionModelBuilder> context)
+        {
+            var sql = new StringBuilder();
+
+            foreach (var (rootEntityType, discriminatorValues) in modelBuilder.Metadata
+                .GetEntityTypes()
+                .GroupBy(e => e.GetRootType())
+                .Where(g => g.Key.GetDiscriminatorProperty() != null && g.Key.GetIsDiscriminatorMappingComplete())
+                .Select(g => (g.Key, g.Select(e => e.GetDiscriminatorValue()))))
+            {
+                var discriminatorProperty = rootEntityType.GetDiscriminatorProperty();
+                sql.Clear();
+
+                sql.Append(_sqlGenerationHelper.DelimitIdentifier(discriminatorProperty.GetColumnName()));
```

Comment:
Column name could be `null`

Reply:
In what cases does this happen? Doesn't this return the default column name calculated from the property name, so always non-null? One day we will have nullability annotations on everything :wink: 

Just skip cases where either the column or table names are null?

Reply:
If the entity type is mapped to view then GetColumnName returns null. GetViewColumnName would return column name for view. Expand further for entity type mapped to SqlQuery and TVF.

Reply:
OK, thanks. I'm pushing a change to skip entities where the discriminator or enum property has no column name. I'm assuming that if GetColumnName returns non-null, GetTableName also returns non-null.

Does that sound right?

Reply:
We don't have `GetViewColumnName` anymore, GetColumnName returns the default column name, but can still return `null` if the property was explicitly mapped to `null` column. Also it might return a different value when called for the specific table:

```C#
var tableIdentifier  = StoreObjectIdentifier.Create(rootEntityType, StoreObjectType.Table);
discriminatorProperty.GetColumnName(tableIdentifier);
```

`tableIdentifier`  could be `null`